src/hw02/Bank.java

```diff
@@ -0,0 +1,156 @@
+package hw02;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * @author Single-eye
+ */
+/**
+ * Bank class replicating the function of an actual bank: its functions includes to open, modify ,
+ * and deletes accounts. *
+ */
+public class Bank {
+
+   // Holds customers accounts with the CustomerID as the key
+   private Map<Integer, Account> bankAccounts = new HashMap<>();
+
+   /**
+    * default constructor
+    */
+   public Bank() {
+
+   }
+
+   /**
+    * add a customer account after checking if an account exist or not
+    *
+    * @param in_Cust_In
+    * @param inAccount
+    */
+   public boolean addAccount(int in_Cust_In, Account inAccount) {

```

Comment:
There's a potential bug here: the value of the in_Cust_In paramter may not be the same as the ID field in the Account class. In other words, the caller could pass the value 1000 as the ID with an Account object that has the ID 2000. This would cause the account to be put into the hashmap with the wrong key, leading to adding and withdrawing money from the wrong account.

I have two suggestions:

1. We could remove the first parameter and get the account ID by calling inAccount.getId(). This resolves the mismatch issue.

2. Don't require the caller to be responsible for creating the account. Just have them pass in the account balance, then create the account in the addAccount method. This resolves the mismatch and also gives the ability to ensure a unique ID, which the caller can't do because it doesn't have access to the hash map.